src/raftstore/store/peer.rs

```diff
@@ -722,7 +735,14 @@ impl Peer {
     pub fn check_stale_state(&mut self) -> StaleState {
         let naive_peer = !self.is_initialized() || self.raft_group.raft.is_learner;
         // Updates the `leader_missing_time` according to the current state.
-        if self.leader_id() == raft::INVALID_ID {
+        // If we haven't heard from any peer in a heartbeat interval, or there is no peers, we're
+        // probably isolated.
+        if self.peer_heartbeats
```

Comment:
What if the peer is a follower?

Reply:
We talked on chat and I will work on improving this based on our conversation. :+1: